mautic/mautic
# Update SpoolTransport.php to set new MessageID for each email

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #11392 



#### Description:

Fixes issue with Outlook not displaying all emails when queued and sent to catch-all / plus forwarding addresses with the same MessageID value

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Setup Mautic to Queue emails (Settings->Email Settings->Queue)
3. Setup Catch-All/Plus Forwarding on your email account (Outlook)
4. Create 3-4 accounts that use the catch-all/plus forwarding rules in Mautic (Ex: test+1@test.com, test+2@test.com, test+3@test.com, test+4@test.com)
5. Create a segment that includes all of these accounts
6. Create a segment based email (hello world template is fine)
7. Send the email to the segment
8. All the accounts should get the email (before this change only one will get the email)




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11400"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
